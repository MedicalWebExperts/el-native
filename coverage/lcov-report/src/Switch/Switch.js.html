<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/Switch/Switch.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/Switch</a> Switch.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { bool, func, object, string } from 'prop-types';
import { Switch as SwitchRN } from 'react-native';
&nbsp;
const propTypes = {
  /**
   * Action to execute when the user toggles the switch.
   */
  onPress: func,
  /**
   * Custom style.
   */
  style: object,
  /**
   * Background color when the switch is turned on. Default: Theme primary color.
   */
  color: string,
  /**
   * The value of the switch. If true the switch will be turned on. Default: false.
   */
  value: bool,
  /**
   * If true the user won't be able to toggle the switch. Default: false.
   */
  disabled: bool,
};
&nbsp;
const defaultProps = {
  onPress: null,
  style: {},
  /**
   * TODO: Replace with theme primary color.
   */
  color: 'lightgreen',
  value: false,
  disabled: false,
};
&nbsp;
/**
 * Boolean input.
 *
 * Requires a onPress callback that updates the `value` prop in order to reflect user actions.
 * If no callback is provided, the component will continue to render the value supplied.
 */
const Switch = props =&gt; (
  &lt;SwitchRN
    style={props.style}
    onValueChange={props.onPress &amp;&amp; typeof props.onPress === 'function' ? props.onPress : <span class="fstat-no" title="function not covered" >()</span> =&gt; {}}
    onTintColor={props.color}
    value={props.value}
    disabled={props.disabled}
  /&gt;
);
&nbsp;
Switch.propTypes = propTypes;
Switch.defaultProps = defaultProps;
&nbsp;
export default Switch;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 17 2018 12:58:56 GMT-0300 (-03)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
