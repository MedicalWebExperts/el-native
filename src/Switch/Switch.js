import React from 'react';
import { bool, func, object, string } from 'prop-types';
import { Switch as SwitchRN } from 'react-native';

const propTypes = {
  /**
   * Action to execute when the user toggles the switch.
   */
  onPress: func,
  /**
   * Custom style.
   */
  style: object,
  /**
   * Background color when the switch is turned on. Default: Theme primary color.
   */
  color: string,
  /**
   * The value of the switch. If true the switch will be turned on. Default: false.
   */
  value: bool,
  /**
   * If true the user won't be able to toggle the switch. Default: false.
   */
  disabled: bool,
};

const defaultProps = {
  onPress: null,
  style: {},
  /**
   * TODO: Replace with theme primary color.
   */
  color: 'lightgreen',
  value: false,
  disabled: false,
};

/**
 * Boolean input.
 *
 * Requires a onPress callback that updates the `value` prop in order to reflect user actions.
 * If no callback is provided, the component will continue to render the value supplied.
 */
const Switch = props => (
  <SwitchRN
    style={props.style}
    onValueChange={props.onPress && typeof props.onPress === 'function' ? props.onPress : () => {}}
    onTintColor={props.color}
    value={props.value}
    disabled={props.disabled}
  />
);

Switch.propTypes = propTypes;
Switch.defaultProps = defaultProps;

export default Switch;
