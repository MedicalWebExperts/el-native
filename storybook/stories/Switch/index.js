import React from 'react';

import { storiesOf } from '@storybook/react-native';
import { View } from 'react-native';

import CenterView from '../CenterView';
import { Switch } from '../../../src';

const customColor = 'aquamarine';
const onPress = () => console.warn('I was pressed');

storiesOf('Switch', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('Default', () => (
    <View>
      <Switch onPress={onPress} />
      <Switch value onPress={onPress} />
      <Switch disabled onPress={onPress} />
      <Switch disabled value onPress={onPress} />
    </View>
  ))
  .add('Custom color', () => (
    <View>
      <Switch color={customColor} onPress={onPress} />
      <Switch value color={customColor} onPress={onPress} />
      <Switch disabled color={customColor} onPress={onPress} />
      <Switch disabled value color={customColor} onPress={onPress} />
    </View>
  ));
