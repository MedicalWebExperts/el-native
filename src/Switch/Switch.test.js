/* global describe it expect jest:true */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Switch from './Switch';

describe('Switch Snapshots', () => {
  it('should render a Switch', () => {
    const component = renderer.create(<Switch />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a Switch toggled on', () => {
    const component = renderer.create(<Switch value />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a Switch disabled', () => {
    const component = renderer.create(<Switch disabled />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a Switch disabled toggled on', () => {
    const component = renderer.create(<Switch disabled value />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a Switch with custom color', () => {
    const component = renderer.create(<Switch color="#FAFAFA" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Interactions', () => {
  it('should execute onPress callback', () => {
    const props = { onPress: jest.fn() };
    const wrapper = shallow(<Switch {...props} />);

    wrapper.prop('onValueChange')();
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });
});
