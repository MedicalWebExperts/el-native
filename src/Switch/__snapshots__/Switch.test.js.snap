// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Snapshots should render a Switch 1`] = `
<RCTSwitch
  disabled={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#2196F3"
  onValueChange={[Function]}
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Object {},
    ]
  }
  thumbTintColor={null}
  tintColor={null}
  value={false}
/>
`;

exports[`Switch Snapshots should render a Switch disabled 1`] = `
<RCTSwitch
  disabled={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#2196F3"
  onValueChange={[Function]}
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Object {},
    ]
  }
  thumbTintColor={null}
  tintColor={null}
  value={false}
/>
`;

exports[`Switch Snapshots should render a Switch disabled toggled on 1`] = `
<RCTSwitch
  disabled={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#2196F3"
  onValueChange={[Function]}
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Object {},
    ]
  }
  thumbTintColor={null}
  tintColor={null}
  value={true}
/>
`;

exports[`Switch Snapshots should render a Switch toggled on 1`] = `
<RCTSwitch
  disabled={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#2196F3"
  onValueChange={[Function]}
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Object {},
    ]
  }
  thumbTintColor={null}
  tintColor={null}
  value={true}
/>
`;

exports[`Switch Snapshots should render a Switch with custom color 1`] = `
<RCTSwitch
  disabled={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#FAFAFA"
  onValueChange={[Function]}
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Object {},
    ]
  }
  thumbTintColor={null}
  tintColor={null}
  value={false}
/>
`;
